example/cumulus-tf/parse_pdr_workflow_unique.asl.json

```diff
@@ -0,0 +1,244 @@
+{
```

Comment:
This workflow is required to *set* the task config for ParsePdr and still remain compatible with a copy of existing tests.   There's a lot of boilerplate here, open to consolidation/refactor considerations on review.

Reply:
Yes - that's supported by adding it to the workflow and referencing the `collection` meta object if they so desire, e.g.  ${meta.collection.meta.<key>}`

With task components we've provided hooks in that way to avoid tying the configuration down to *only* collection, it can be per-collection, or per workflow, or some other value in the message meta, depending on their need, as the components shouldn't care about where they're getting the value from unless there's a good reason for it to be prescriptive.

Reply:
Hmm, that's a fair point.   To some degree I was erring on the side of not buying that in a collection configuration but making it an explicit workflow choice to avoid false positives.     Fair feedback that's overkill 🤔

Reply:
I went ahead and updated that, and made a not-insignificant change to typings and/or spec test assertions that should hopefully look better.  a6d3b8180d6

Reply:
@jennyhliu  one unexpected consequence of this is that (and this likely doesn't map to how users are utilizing the workflows) it highlighted that CMA/step function treat undefined keys as nullish - meaning the task needs to accept nullish config params for these values if we're going to be using it this way.  We should probably consider this pattern in other places as well. 

Good discussion and I think this is a good improvement to parse-pdr: 6fa72ae508b

Reply:
Yeah, I tested the typings are necessary